test/framework/clusterresourceset_helpers.go

```diff
@@ -129,23 +131,21 @@ func WaitForClusterResourceSetToApplyResources(ctx context.Context, input WaitFo
 		Expect(input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: input.Cluster.Name, Namespace: input.Cluster.Namespace}, binding)).To(Succeed())
 
 		for _, resource := range input.ClusterResourceSet.Spec.Resources {
-			if resource.Kind == string(addonsv1.SecretClusterResourceSetResourceKind) {
-				secret := &corev1.Secret{}
-				err := input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: input.ClusterResourceSet.Namespace}, secret)
-				if err == nil {
-
-					if !binding.Spec.Bindings[0].IsApplied(resource) {
-						return false
-					}
-				}
-			} else if resource.Kind == string(addonsv1.ConfigMapClusterResourceSetResourceKind) {
-				configMap := &corev1.ConfigMap{}
-				err := input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: input.ClusterResourceSet.Namespace}, configMap)
-				if err == nil {
-					if !binding.Spec.Bindings[0].IsApplied(resource) {
-						return false
-					}
-				}
+			var configSource runtime.Object
+
+			switch resource.Kind {
+			case string(addonsv1.SecretClusterResourceSetResourceKind):
+				configSource = &corev1.Secret{}
+			case string(addonsv1.ConfigMapClusterResourceSetResourceKind):
+				configSource = &corev1.ConfigMap{}
+			}
+
+			if err := input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: input.ClusterResourceSet.Namespace}, configSource); err != nil {
+				continue
```

Comment:
@sedefsavas I'm not overly familiar with these tests, so I wasn't sure if it was intentional that we continued through the loop when the Get for the related Secret/ConfigMap returned an error, so I kept the existing behavior. I'm wondering if this should return `false` here instead, though.

Reply:
If the resource is missing, CRS will not requeue but retry at each reconcile, because this is not an error. So, we are only interested in seeing the resources that exist to be applied by CRS.

Reply:
Can we add a comment to that effect?

Reply:
This test is going to be removed totally with this PR: https://github.com/kubernetes-sigs/cluster-api/pull/3332

Reply:
Are we planning to merge that in v0.3.8, or later? If later, I'd prefer to have a comment just in case it stays around for longer

Reply:
There are changes in providers as well and all need to go in at once.
We are waiting CAPA's next release, there is a change there we need to wait.